### PR TITLE
Finance form improvements

### PR DESCRIPTION
**Pull request by ciliakas**
Improvements on the finance form:
- Added more data on the side panel
- Renamed some methods in DataTableConverter and deleted unused ones

### CODE REVIEW

**Review by MatasLa — APPROVED** [review]
Tested, it seems to be working fine and data is shown correctly.

File: SmartSaver/DataManagement/DataTableConverter.cs [hunk 1]
```diff
@@ -10,88 +10,57 @@ public class DataTableConverter
...
                 }
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(entryType), entryType, null);
             }
             return dt;
         }
 
         /*Takes list from DataFilter and creates income or expenses DataTable, used to get DataTable of filtered List*/
-        public DataTable CustomTable(List<DataEntry> customList)
+        public static DataTable GenerateEntryTable(List<DataEntry> customList)
```
Comment (Stokalas) [inline SmartSaver/DataManagement/DataTableConverter.cs:44]: Just a question: Why was this changed to be static?

File: SmartSaver/forms/finances/budget/FinanceForm.cs [hunk 2]
```diff
@@ -275,24 +284,17 @@ private void DisplayTable()
...
             dataGridView.Columns["Amount"].DefaultCellStyle.Format = "c";
             dataGridView.Columns["Date"].DefaultCellStyle.Format = "d";
         }
 
-
         private bool DataTableErrorCheck()
         {
             if (dataGridView.Columns["ID"] is null)
             {
                 ExceptionHandler.Log("Data Table Generation Error: NO ID COLUMN");
```
Comment (Stokalas) [inline SmartSaver/forms/finances/budget/FinanceForm.cs:295]: Maybe these Error strings should be declared at the top of the class as readonly strings?

**Review by Stokalas — COMMENTED** [review]
While everything seems to be clean and well done, I had few questions, which I asked in comments of code.

File: SmartSaver/DataManagement/DataTableConverter.cs (hunk 1 above)
Comment (ciliakas) [reply SmartSaver/DataManagement/DataTableConverter.cs:44]: Since this method is not using any of the variables declared in the class (in our case it's only the Data variable) as resharper suggests I changed it to static.

File: SmartSaver/forms/finances/budget/FinanceForm.cs (hunk 2 above)
Comment (ciliakas) [reply SmartSaver/forms/finances/budget/FinanceForm.cs]: Good idea, that would make more sense. However after looking at this piece of code once again, I realized that this error testing was not necessary, because the table that I use always generate these headers, so there should never be a null reference exception here.

**Review by Stokalas — APPROVED** [review]
